### ISSUE TITLE
No, the Copilot key cannot be remapped.

### ISSUE THREAD

**Issue by randyrants** [randyrants/sharpkeys#560]
Hi SharpKeys fans!

Sorry to report that the new Copilot key that is appearing on new PC's is not remappable by SharpKeys.  

The Copilot key's scancode is a combination of Ctrl + Windows + F23 and the Windows technology the SharpKeys exposes only supports remapping an entire key and not a modified combination of the key.

**Comment by pavi2410** [comment]
PowerToys' Keyboard Manager tool is able to remap the Copilot key but that doesn't work for all apps consistently.

https://www.reddit.com/r/olkb/s/tPXmeJ0UGf

Also note:
> It seems that
> 
> R Ctrl + R Shift + P works
> 
> but
> 
> R Ctrl + L Shift + P doesn't

**Comment by Ronjar** [comment]
Remapping the F23 Key (00_6E) worked for me, although I have no idea why.
Worked on a Lenovo Ideapad 5, maybe others could also try and report.

**Comment by DreymaR** [comment]
You know, that kind of makes sense. If the Copilot key sends Ctrl+Shift+F23 and you remap F23, then in theory you'd get Ctrl+Shift+<new key>. If that's okay, then it should work.

Or are you saying that the Ctrl+Shift gets removed somehow?

This reminds me of remapping the RAlt, which works as a remap for AltGr even though AltGr technically sends LCtrl+RAlt.

**Comment by randyrants** [comment]
Well, that's an interesting twist.  

Yes, it does make sense, and I just tried it on a recent Surface Laptop (Intel-based) - it works fine.

However, if there are other apps/keyboards using F23 for whatever reason, it will impact that functionality as well, so it's a mixed bag.  I think it's a slam dunk on laptops, which will likely never have an F key above 12, but not for someone with that 122 key IBM Model M, which has a dedicated F23 key.

**Comment by DreymaR** [comment]
Such a keyboard won't have a Copilot key! However, I suspect you could use Ctrl+Shift+F23 to get its functionality then.

**Comment by myusrn** [comment]
> Remapping only the F23 Key (00_6E) worked for me, although I have no idea why. Worked on a Lenovo Ideapad 5, maybe others could also try and report.

> Well, that's an interesting twist.
> 
> Yes, it does make sense, and I just tried it on a recent Surface Laptop (Intel-based) - it works fine.
> 
> However, if there are other apps/keyboards using F23 for whatever reason, it will impact that functionality as well, so it's a mixed bag. I think it's a slam dunk on laptops, which will likely never have an F key above 12, but not for someone with that 122 key IBM Model M, which has a dedicated F23 key.

I tried this on msft surface laptop 7 (snapdragon x elite based) and it did not work. Specifically, I tried mapping "Function: F23 (00_6E)" to "Special: Right Windows (E0_5C)", so i'd have keyboard symmetry on left and right side of space bar, and after reboot no luck. 

Previously with other keyboards like msft sculpt desktop ergnomic wireless keyboard and lenovo laptop i was able to remap their context menu "Special: Application (E0_5D)" and print screen "Special: PrtSc (E0_37)" keys occupying same location as copilot key on new copilot+pc keyboards to the "Special: Right Windows (E0_5C)" key, again to produce keyboard symmetry, and it worked in those cases.

**Comment by randyrants** [comment]
Copilot key shares a home with the Application key.  After remapping, make sure that Fn-Lock is off:

- With Fn-Lock off, it's the Copilot key
- With Fn-Lock on, it's the Application key

If you want it to be a Windows key no matter what state Fn is in, remap the F23 and the Application keys to Right Windows.

**Comment by myusrn** [comment]
> Copilot key shares a home with the Application key. After remapping, make sure that Fn-Lock is off:
> 
> * With Fn-Lock off, it's the Copilot key
> * With Fn-Lock on, it's the Application key
> 
> If you want it to be a Windows key no matter what state Fn is in, remap the F23 and the Application keys to Right Windows.

Tried it again and did a full reboot to ensure it wasn't just a logout / log back in related issue. Ensured function key is off and it does nothing. Turn function key on and the application context menu behavior works as expected. Wondering if running windows on arm or the latest 24H2 26100.2033 install is causing the unexpected result that didn't affect others.

**Comment by DreymaR** [comment]
Are you sure it works fully though? If the Copilot key really sends Ctrl+Shift+F23, then remapping F23 to something else should keep the Ctrl+Shift modifiers for the remapped key.

Ctrl(+Shift)+Windows does nothing as far as I can tell. Shift+Ctrl may work like the unshifted version, but not always. Are you saying that you get the unshifted version with that remap? If so, may the Shift be lost but not the Ctrl?

**Comment by myusrn** [comment]
Hi @DreymaR thanks for the response. 

Yes I might expect that remapping right Ctrl key worked since Ctrl+Shift+Ctrl gets interpreted under most desktop state scenarios as simply a Ctrl key press. I tested Ctrl+Shift+C[opy] using actual left side Ctrl key on my device and I get the same result as Ctrl+C[opy]. Couldn't think of a way to test Ctrl+Shift+Ctrl+C[opy] to do a true test of what the remapping does though I doubt there is a way to send two Ctrl key inputs at once.

Conversely Ctrl+Shift+Left/Right-Windows key doesn't get interpreted as just the Windows key press. I tested Ctrl+Shift+Left-Windows key using the actual left side Windows key on my device and it does the same thing that that having remapped the F23 key to Right-Windows key was doing, which was nothing.

**Comment by randyrants** [comment]
In so far as I know, Copilot's multi-key short cut is WIN+C and the Copilot key itself is F23.

When Windows remaps a key, it remaps the entire key, so it remaps CTRL or C but cannot remap CTRL+C.  This is done at a very low level within Windows, and why combinations after the remapping are effective.  I think.  

Power Toys (and AutoHotKey) is using an additional bit of code that is an active remapper, so it can do modified keys, _but_ there's no guarantee that when combinations of modified keys may be hard to predict, similar to how swapping keys in SharpKeys may take some trial and error.  I mean, think about sending in three modified keys into the Power Toys module: depending on how they get queued up it could end up weird.  And there's no order of operations for keyboard presses like there is for math.  Like when you look at 3 +7 * 10 there are rules to guarantee the answer is 73 and not 100 but for scancodes typed, the queue may get filled differently at different times..

**Comment by myusrn** [comment]
Hi @randyrants thanks for clarifications and additional details. Ultimately it would be best if I could just use a SharpKeys applied physical key remapping to get that copilot key doing something more useful like a right side CTRL given keyboard doesn't include that. 

Especially given if someone is wanting keyboard shortcut for launching copilot app, which appears to just be a native client app wrapper of web browser control, they can just use the WIN+C[opilot] keyboard shortcut that covers that.

Not sure what else I might experiment with since I did try using SharpKeys to remap copilot key [ / F23 ] to CTRL and it works in apps like notepad, where it seems the Win (Left)  + Shift (Left) modifiers being included doesn't make a difference, but not in apps like outlook and teams, where it seems those modifiers do block interpretation of a simple CTRL key for use as a C, V or X key modifier. 

I noticed that unlike SharpKeys key remapping which works right away when you login to the shell the PowerToys keyboard manager remapping of Win (Left) + Shift (Left) + F23 on my surface 7 snapdragon X arm64 device takes 30sec of so to take effect. Essentially however much time it takes for system from a new shell logging session to get around to launching the PowerToys system tray daemon.

**Comment by randyrants** [comment]
Again, SharpKeys is not remapping anything: it is simply an app that exposes the Registry value that Windows uses to remap the key.  This is why there needs to be a reboot, while Power Toys and AutoHotKeys are actively loading an additional bit of code to remap the key above Windows.

As I've already called out that the Copilot key's code is supposed to be CTRL+WIN+F23, Windows was never expected to remap this key.  That you can sort of workaround it by remapping F23, I don't think it's going to remap that well, because it's actually going to be CTRL+WIN+whatever-key-you-remapped.

**Comment by pavi2410** [comment]
> PowerToys' Keyboard Manager tool is able to remap the Copilot key but that doesn't work for all apps consistently.
> 
> https://www.reddit.com/r/olkb/s/tPXmeJ0UGf
> 
> Also note:
> 
> > It seems that
> > R Ctrl + R Shift + P works
> > but
> > R Ctrl + L Shift + P doesn't

Note to self: See the screenshot below 😅 

![Image](https://github.com/user-attachments/assets/d95e3cb7-55d6-4acb-8507-592e8abd441d)

---

Yesterday, I had to reset my PC and tripped on this issue yet again! I am able to successfully remap Copilot key by using PowerToys.

On PowerToys's Keyboard Manager tool, use "Remap a shortcut" instead of "Remap a key" and remap as shown below:
    
      Win (Left) + Shift (Left) + F23 -> Ctrl (Right)

**Comment by Arxareon** [comment]
Unfortunately it still doesn't work well. Remapping like what can bee seen on the image pavi posted but to left Win key instead of Ctrl, I get weird behavior like the pressing Win + V and the Start menu opening up instead of the clipboard appearing because I was too fast. Or, the clipboard window not staying open and disappearing when I release the Copilot key reassigned as the Win key.
It simply just sucks.

We need to do something better, not sure what we can..

**Comment by Baker7** [comment]
Wow!  Tried the Ronjar suggestion above!!! IT WORKED.

I used Sharpkeys to Remap F23 to Right Shift ===> Now the co-pilot key does nothing!

**Comment by myusrn** [comment]
> Wow! Tried the Ronjar suggestion above!!! IT WORKED.
> 
> I used Sharpkeys to Remap F23 to Right Shift ===> Now the co-pilot key does nothing!

While i find Sharpkeys great and used it for many years now with the powertoys store app's keyboard manager i find it just easier to setup multiple key shortcuts.  Specifically ones that point to the positional number of consumer/personal copilot and business/work m365 copilot and throw in an outlook and teams client shortcut as well using following task bar app icon ordering and powertoys keyboard manager shortcuts for opening and minimizing them.

<img width="573" height="75" alt="Image" src="https://github.com/user-attachments/assets/565de611-0966-4815-a0c8-32b9fb112845" />

<img width="937" height="497" alt="Image" src="https://github.com/user-attachments/assets/904632ea-7149-4964-8b1b-5e9f64444f9b" />

**Comment by Baker7** [comment]
PowerToys turns on tons of extra things I don't want.  And it needs to permanently run in the background.

I had used Power Toys to try to disable the Co-Pilot key, if I recall it still sometimes worked.  But all the extra stuff in Power Toys was active as well and some of them did things I did not want.

**Comment by myusrn** [comment]
Yes it comes with many options and quite a few enabled by default that can easily be disabled in settings. I just run the custom window zones / position + sizing experience and keyboard manager shortcuts. Occasionally will enable the caffeine option that lets you force system to stay powered on for specific period or time before allowing sleep behaviors to kick in

**Comment by Baker7** [comment]
> Yes it comes with many options and quite a few enabled by default that can easily be disabled in settings. I just run the custom window zones / position + sizing experience and keyboard manager shortcuts. Occasionally will enable the caffeine option that lets you force system to stay powered on for specific period or time before allowing sleep behaviors to kick in

I don't want 2 dozen extra features enabled by default and then have to spend an extended period of time looking through the numerous menu items for each of the categories.

I'm not trying to rain on PowerToys, you seem to like it.  But please understand, there are some like myself that find PowerToys extremely disappointing and I found the mouse/window/user interface side effects very unappealing.

**Comment by myusrn** [comment]
i get it everyone has preferences for simple vs bloated solutions to just the functionality they want to light up on their workstations

**Comment by Rast1234** [comment]
Hey, i had a deep dive to figure out what can be done about unwanted keys, and did a writeup here: https://github.com/Rast1234/copilot_key_on_my_keyboard

TL;DR sharpkeys is the least problematic way to go about it, and you can even restore some missing key, sort of.

**Comment by faq0** [comment]
Although direct remapping of `LShift+LWin+RCtrl` is impossible in Powertoys, it seems like remapping `LShift+LWin+RCtrl+*` combinations are possible.

Just remapped a few combinations I use daily. Especially `Ctrl+P`, which is way too hard to reach using one hand.

<img width="1394" height="649" alt="Image" src="https://github.com/user-attachments/assets/93bf5c8e-1292-4c12-816d-ad496996afaa" />

**Comment by carehart** [comment]
Hey @randyrants, might you consider tweaking your opening statement to add an update, indicating that some people ARE finding success remapping it solely as f23? I get that there are technical reasons that your title statement and initial post is otherwise still true--and that this won't work for everyone. 

But some folks being guided to this issue (via searching, etc.) may not read the entire thread and all comments to discern where remapping this key alone MAY well work for them.

**Comment by randyrants** [comment]
Nope.  I don't want the responsibility if there's unforeseen negative behavior by a partial solution that is "lucky" rather than "good."

Given recent articles from Microsoft on Copilot overall, they may take action on this topic, but I'm speculating on that.

**Comment by rthoma** [comment]
I don't know anything about how this works, but is it possible that pressing the copilot key does fire a single scancode that gets mapped to the Win + Shift + f23 combination in software? I found a response on Microsoft Learn that suggests the copilot key fires HID usage ID 0x0D8. I have no idea whether that is accurate, but is there a way to test the effect of remapping that scancode to an arbitrary key on a device with a copilot key? Forgive me if this is far-fetched. https://learn.microsoft.com/en-us/answers/questions/3877536/what-if-i-need-to-add-copilot-key-in-my-keyboard-w

**Comment by randyrants** [comment]
Cannot. Be. Done. With. SharpKeys.

HID is a Human Interface Device code, which has nothing to do with how Windows remaps keys, so it does not help.  WIN+C uses two scan codes.  WIN+SHIFT+F23 uses three scan codes, which is why there might be a random - aka lucky - chance of Windows intercepting the F23 and sending it elsewhere, preventing Copilot from launching, but again: I'm not adding a partial solution that is "lucky."

The suggestion in the Microsoft article is exactly what I lead with, when asked about the Copilot key: use PowerToys or AutoHotKeys.
